server/utils/bookings/index.ts

```diff
@@ -179,33 +184,34 @@ export const generateConflictBespokeError = (
   const { detail } = err.data as { detail: string }
   const { conflictingEntityId, conflictingEntityType } = parseConflictError(detail)
 
-  const title =
-    datesGrammaticalNumber === 'plural'
-      ? 'This bedspace is not available for the dates entered'
-      : 'This bedspace is not available for the date entered'
+  const title = (
+    conflictingEntityType === ConflictingEntityType.lostBed
+      ? 'Out of service bed record cannot be created for the $date$ entered'
+      : 'This bedspace is not available for the $date$ entered'
+  ).replace('$date$', datesGrammaticalNumber === 'plural' ? 'dates' : 'date')
 
   const link =
-    conflictingEntityType === 'lost-bed' && bedId
-      ? `<a href="${paths.lostBeds.show({
-          premisesId,
-          bedId,
-          id: conflictingEntityId,
-        })}">existing lost bed</a>`
+    conflictingEntityType === ConflictingEntityType.lostBed && bedId
+      ? `<a href="${paths.v2Manage.outOfServiceBeds.show({ premisesId, bedId, id: conflictingEntityId, tab: 'details' })}">existing out of service beds record</a>`
       : `<a href="${paths.bookings.show({
           premisesId,
           bookingId: conflictingEntityId,
         })}">existing booking</a>`
-
   const message = datesGrammaticalNumber === 'plural' ? `They conflict with an ${link}` : `It conflicts with an ${link}`
 
   return { errorTitle: title, errorSummary: [{ html: message }] }
 }
 
 const parseConflictError = (detail: string): ParsedConflictError => {
-  const detailWords = detail.split(' ')
-  const conflictingEntityId = detailWords[detailWords.length - 1]
-  const conflictingEntityType = detail.includes('Lost Bed') ? 'lost-bed' : 'booking'
-
+  /**
+   *  Return the entity type and id by parsing an error detail string
+   *  @param detail - string is text containing the entity id at the end preceded by ': '
+   *    e.g. "An out-of-service bed already exists for dates from 2024-10-01 to 2024-10-14 which overlaps with the desired dates: 220a71da-bf5c-424d-94ff-254ecac5b857"
+   */
+  const matchResult = /([0-9a-z-]*)$/.exec(detail)
+  const conflictingEntityId = matchResult && matchResult[1]
+  const isOutOfServiceBedEntity = /(?:lost bed)|(?:out-of-service bed)/i.test(detail)
+  const conflictingEntityType = isOutOfServiceBedEntity ? ConflictingEntityType.lostBed : ConflictingEntityType.booking
```

Comment:
This seems complex, and we shouldn't need to handle the 'Lost bed' scenario anymore.

CAS3 simply splits the string by `:` to parse the entity ID, then searches for a simpler match in the detail:

```suggestion
  /**
   *  Return the entity type and id by parsing an error detail string
   *  @param detail - string is text containing the entity id at the end preceded by ': '
   *    e.g. "An out-of-service bed already exists for dates from 2024-10-01 to 2024-10-14 which overlaps with the desired dates: 220a71da-bf5c-424d-94ff-254ecac5b857"
   */
  const [ message, conflictingEntityId ] = details.split(':').map(trim)
  const conflictingEntityType = message.includes('out-of-service bed') ? ConflictingEntityType.lostBed : ConflictingEntityType.booking
```

With the comment above about not using enums, this last line would then return either `lost-bed` or `booking` instead of the enum syntax.

Reply:
I'm not 100% convinced - mainly because I like regex so much. but I've changed to use your approach.